applications/system/js_app/types/gui/index.d.ts

```diff
@@ -3,11 +3,28 @@ import type { Contract } from "../event_loop";
 type Properties = { [K: string]: any };
 
 export declare class View<Props extends Properties> {
+    /**
+     * Assign value to property by name
+     * @param property Name of the property
+     * @param value Value to assign
+     */
     set<P extends keyof Props>(property: P, value: Props[P]): void;
+    /**
+     * Check if property is available
+     * @param name Name of the property
+     */
+    hasProperty(name: string): boolean;
```

Comment:
I think it's possible to compute the return value of this function at compile time with something to the tune of:

```ts
hasProperty<K extends string>(name: K): K extends keyof Props ? true : false;
```

Not sure why this API method would exist at all, given that TypeScript will check things :thinking:

Reply:
Typescript is useful to the developer making the script, but maybe they want to make their scripts backwards or forwards compatible, or work across firmwares. Without this, someone adding a custom prop locks scripts to their firmware. With this, the script can check and work on all firmwares.

Reply:
For example, the keyboard we have in momentum supports characters that would break file paths, so they're disabled by default but can be enabled with an api call. I want to add a prop for this, but now any script using this will not work on other firmwares. I'd rather have a way to check availability than make a walled garden around me.

We could go onto "why not add this keyboard functionality to ofw?", and that is a valid point, but besides the objective of this function. This is just an example, there's bound to be more cases in the future, maybe a prop is added in ofw dev but isn't in ofw stable yet, no need to wait to update your script.

Reply:
I'm thinking that there needs to be a more generic mechanism for detecting discrepancies between the JS API that a script expects, and the JS API that the firmware provides. This is one of the problems that I'm currently trying to solve while making a full-fledged JS SDK.

I'm thinking we should introduce a JS SDK major-minor versioning system and check whether the two are compatible on script startup. Kinda like how native apps do this already.

Reply:
@Willy-JL talking about api versioning, I think we can find compromise:
- API versioning in JS got different nature, not like api versioning in API symbols, it should be much easier to maintain it
- We'll have checkSdk() but as an optional canary
- If you call it we'll check version and ask user what to do if version doesn't match
- If your script doesn't contain it, then no checks is done and you'll crash later
- I'd like to see all of us cooperating on JS API: keep it consistent across all forks, simplifying portability between different firmware flavors. It make sense to expose additional API to query which firmware flavor we are running at

Reply:
alright that sounds like it could work, ill remove hasProperty() and hold off on implementing this extra prop i wanted until #3963 is consolidated